builtin/providers/random/resource_id.go

```diff
@@ -4,71 +4,88 @@ import (
 	"crypto/rand"
 	"encoding/base64"
 	"encoding/hex"
-	"fmt"
+	"errors"
 	"math/big"
 
+	"github.com/hashicorp/errwrap"
 	"github.com/hashicorp/terraform/helper/schema"
 )
 
 func resourceId() *schema.Resource {
 	return &schema.Resource{
 		Create: CreateID,
-		Read:   stubRead,
-		Delete: stubDelete,
+		Read:   schema.Noop,
+		Delete: schema.RemoveFromState,
 
 		Schema: map[string]*schema.Schema{
-			"keepers": &schema.Schema{
+			"keepers": {
 				Type:     schema.TypeMap,
 				Optional: true,
 				ForceNew: true,
 			},
 
-			"byte_length": &schema.Schema{
+			"byte_length": {
 				Type:     schema.TypeInt,
 				Required: true,
 				ForceNew: true,
 			},
 
-			"b64": &schema.Schema{
+			"b64": {
+				Type:       schema.TypeString,
+				Computed:   true,
+				Deprecated: "Use b64_url for old behavior, or b64_std for standard base64 encoding",
+			},
+
+			"b64_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"b64_std": {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
 
-			"hex": &schema.Schema{
+			"hex": {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
 
-			"dec": &schema.Schema{
+			"dec": {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
 		},
 	}
 }
 
-func CreateID(d *schema.ResourceData, meta interface{}) error {
+func CreateID(d *schema.ResourceData, _ interface{}) error {
 
 	byteLength := d.Get("byte_length").(int)
 	bytes := make([]byte, byteLength)
 
 	n, err := rand.Reader.Read(bytes)
 	if n != byteLength {
-		return fmt.Errorf("generated insufficient random bytes")
```

Comment:
No need to cover this here, but noting that this catches the global rand source and should probably be reworked to use a local one at some point.